### PR TITLE
Modified Asset Cloudlet protocol

### PR DESCRIPTION
**Pull request by nicolatimeus**
See https://github.com/eclipse/kura/wiki/ASSET-V1-MQTT-Namespace

Signed-off-by: nicolatimeus <nicola.timeus@eurotech.com>

### CODE REVIEW

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/serialization/SerializationConstants.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ *******************************************************************************/
+
+package org.eclipse.kura.internal.asset.cloudlet.serialization;
+
+public class SerializationConstants {
+
```
Comment (amitjoy) [inline kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/serialization/SerializationConstants.java]: Add a private constructor or make it an interface. But I would still prefer using enum as it serves the exact need for defining constants and enums represent similar values as well.

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/serialization/response/AbstractResponse.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.eclipse.kura.internal.asset.cloudlet.serialization.SerializationConstants;
+import org.eclipse.kura.localization.LocalizationAdapter;
+import org.eclipse.kura.localization.resources.AssetCloudletMessages;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+
+public class AbstractResponse {
```
Comment (amitjoy) [inline kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/serialization/response/AbstractResponse.java]: Make it abstract

File: kura/org.eclipse.kura.asset.cloudlet.provider/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -7,7 +7,8 @@ Bundle-Vendor: Eclipse Kura
 Bundle-License: Eclipse Public License v1.0
 Bundle-Category: Asset-Driver Management
 Bundle-RequiredExecutionEnvironment: JavaSE-1.8
-Import-Package: org.eclipse.kura;version="[1.1,2.0)",
+Import-Package: com.eclipsesource.json;version="0.9.0",
+ org.eclipse.kura;version="[1.1,2.0)",
```
Comment (MMaiero) [inline kura/org.eclipse.kura.asset.cloudlet.provider/META-INF/MANIFEST.MF]: This import has to be in the form [1.3,2.0)

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java [hunk 4]
```diff
@@ -194,272 +117,196 @@ protected synchronized void deactivate(final ComponentContext componentContext)
...
+        JsonArray request;
+
+        try {
+            request = getRequestJsonArray(reqPayload);
+        } catch (Exception e) {
+            respPayload.setResponseCode(KuraResponsePayload.RESPONSE_CODE_BAD_REQUEST);
+            return;
+        }
+
+        if (request == null || request.size() == 0) {
```
Comment (MMaiero) [inline kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java]: Maybe it's better to check if isEmpty()?

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java [hunk 5]
```diff
@@ -194,272 +117,196 @@ protected synchronized void deactivate(final ComponentContext componentContext)
...
-
-        final String channelDelim = "_";
-        Set<String> channelNames = null;
-        if (channelName.contains(channelDelim)) {
-            channelNames = CollectionUtil.newHashSet(Arrays.asList(channelName.split(channelDelim)));
-            channelNames.removeAll(Collections.singleton(""));
+        JsonArray request;
+
+        try {
+            request = getRequestJsonArray(reqPayload);
```
Comment (MMaiero) [inline kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java]: maybe we can call it parseRequest?

File: kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java [hunk 6]
```diff
@@ -194,272 +117,196 @@ protected synchronized void deactivate(final ComponentContext componentContext)
...
             }
-            if (channelRecords != null) {
-                this.prepareResponse(respPayload, channelRecords);
+            response = new ChannelOperationResponse();
+            for (ReadRequest readRequest : readRequests) {
+                readAsset(readRequest.getAssetName(), readRequest.getChannelNames(), response);
             }
         }
+
+        respPayload.setBody(response.getSerialized().toString().getBytes(StandardCharsets.UTF_8));
```
Comment (MMaiero) [inline kura/org.eclipse.kura.asset.cloudlet.provider/src/main/java/org/eclipse/kura/internal/asset/cloudlet/AssetCloudlet.java]: Maybe this `response.getSerialized().toString().getBytes(StandardCharsets.UTF_8)` can be done in a specific method in `ChannelOperationResponse `

**Pull request merged**